apps/application/flow/workflow_manage.py

```diff
@@ -676,9 +676,17 @@ def get_next_node_list(self, current_node, current_node_result):
                             self.get_node_cls_by_id(edge.targetNodeId, self.get_up_node_id_list(edge.targetNodeId)))
         else:
             for edge in self.flow.edges:
-                if edge.sourceNodeId == current_node.id and self.dependent_node_been_executed(edge.targetNodeId):
-                    node_list.append(
-                        self.get_node_cls_by_id(edge.targetNodeId, self.get_up_node_id_list(edge.targetNodeId)))
+                if edge.sourceNodeId == current_node.id:
+                    next_node = [node for node in self.flow.nodes if node.id == edge.targetNodeId]
+                    if len(next_node) == 0:
+                        continue
+                    if next_node[0].properties.get('condition', "AND") == 'AND':
+                        if self.dependent_node_been_executed(edge.targetNodeId):
+                            node_list.append(
+                                self.get_node_cls_by_id(edge.targetNodeId, self.get_up_node_id_list(edge.targetNodeId)))
+                    else:
+                        node_list.append(
+                            self.get_node_cls_by_id(edge.targetNodeId, self.get_up_node_id_list(edge.targetNodeId)))
         return node_list
 
     def get_reference_field(self, node_id: str, fields: List[str]):
```

Comment:
This code snippet appears to implement a method that determines which nodes should be processed next based on certain conditions. Here are some potential problems and optimizations:

**Potential Issues:**

1. **List Comprehension in Loop:** 
   - The line `next_node = [node for node in self.flow.nodes if node.id == edge.targetNodeId]` iterates over all nodes once per iteration of the main loop. This is unnecessary when you only need one target node from each connection.

2. **Multiple Returns:**
   - If multiple nodes have the same `targetNodeId`, this code will append them all, but it won't stop after the first match. Consider adding a flag to break out of the function once a match is found to optimize performance for cases where duplicates exist.

3. **Condition Handling:**
   - The code checks whether the condition property exists before accessing it (`if next_node[0].properties.get('condition', "AND")`). However, if `condition` doesn't exist, it will use `"AND"` as default, assuming every condition has been met. Make sure this default behavior aligns with your needs.

4. **Code Reusability:**
   - The function `get_next_node_list` could benefit from reusing logic in helper functions like checking dependencies and getting node classes by ID.

**Suggestions:**

1. **Simplify Condition Check:**
   ```python
   if edge.sourceNodeId == current_node.id:
       # Determine if condition matches OR/AND rule
       if self._should_process_and_rule(node_id, edge.targetNodeId):
           node_list.append(self.get_node_cls_by_id(edge.targetNodeId, self.get_up_node_id_list(edge.targetNodeId)))
   ```

2. **Use Helper Function for Node Retrieval:**
   Create a helper function `_get_target_nodes` that fetches and returns target nodes directly without iterating multiple times per edge.

3. **Add Debugging Information:**
   Insert print statements or logging to trace execution flow and identify bottlenecks more easily.

Overall, improving efficiency and ensuring robust conditional handling would make the code cleaner and potentially faster under certain scenarios.